includes/User.php

```diff
@@ -2169,6 +2167,8 @@ public function invalidateCache() {
 					__METHOD__ );
 			}
 
+			$dbw->commit();
```

Comment:
@macbre that's a good point. The problem for now is that attributes are being stored in the same `mOptions` array as preferences which uses a different service. So even if we checked the attribute service to see when the last "user touched" value was, we'd have to do the same for the preference service, and even then we still have user "flags" which are still stored in MW which have yet to be split out into their own service.

An approach to address this would probably be to split this user data into it's components in the user object (eg, have `attributes`, `preferences`, and `flags` in their own arrays). That way we could reference the corresponding service when checking if cache is stale.

Part of the ground work for that has been done with the creation of the `User::get(Attribute|Preference|Flag)` and `User::set(Attribute|Preference|Flag` methods, but all user data still goes through the `mOptions` variable. Once we split that apart more, we can (and probably should) use the service as the source of truth for last updated.